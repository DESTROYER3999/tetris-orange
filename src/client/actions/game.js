import socket from "../services/socket-api";
import {
  KEY_PRESSED,
  UPDATE_SPECTRES,
  UPDATE_GAME_MODE
} from "../../constants/actionTypes";

export const handleKeyPress = e => {
  socket.emit(KEY_PRESSED, { code: e.code });
};

export const updateSpectres = spectres => ({
  type: UPDATE_SPECTRES,
  spectres
});

export const updateGameMode = mode => {
  return () => {
    socket.emit(UPDATE_GAME_MODE, mode);
  };
};
