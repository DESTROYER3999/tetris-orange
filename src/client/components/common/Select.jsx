import React from "react";
import Select from "@material-ui/core/Select";
import FormHelperText from "@material-ui/core/FormHelperText";
import FormControl from "@material-ui/core/FormControl";
import OutlinedInput from "@material-ui/core/OutlinedInput";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  formControl: {
    minWidth: 150,
    marginRight: "10px",
    color: "white",
    "& .MuiOutlinedInput-root": {
      "& fieldset": {
        borderColor: "white"
      },
      "&:hover fieldset": {
        borderColor: "white"
      },
      "&.Mui-focused fieldset": {
        borderColor: "white"
      },
      "&.Mui-disabled fieldset": {
        borderColor: "white"
      }
    }
  },
  label: {
    padding: "0.5vh",
    fontSize: "2vh",
    color: "white"
  },
  select: {
    color: "white",
    "& .MuiSelect-icon": {
      color: "white"
    },
    "& .MuiSelect-root": {
      fontFamily: "ArcadeClassic"
    }
  }
}));

const SelectMenu = props => {
  const classes = useStyles();

  return (
    <FormControl className={classes.formControl}>
      <label className={classes.label}>{props.label}</label>
      <Select className={classes.select} {...props} input={<OutlinedInput />}>
        {props.children}
      </Select>
      {props.helperText ? (
        <FormHelperText style={{ height: "2vh" }}>
          {props.helperText}
        </FormHelperText>
      ) : null}
    </FormControl>
  );
};

export default SelectMenu;
