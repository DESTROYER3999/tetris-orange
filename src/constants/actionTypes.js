export const JOIN_ROOM = "JOIN_ROOM";
export const UPDATE_ROOM = "UPDATE_ROOM";
export const LEAVE_ROOM = "LEAVE_ROOM";
export const LOBBY_ROOM = "LOBBY_ROOM";

export const NEW_PLAYER = "NEW_PLAYER";
export const DISCONNECT_PLAYER = "DISCONNECT_PLAYER";
export const UPDATE_PLAYER = "UPDATE_PLAYER";
export const UPDATE_PLAYERS_LIST = "UPDATE_PLAYERS_LIST";
export const DELETE_PLAYER = "DELETE_PLAYER";

export const HANDLE_HASH = "HANDLE_HASH";

export const UPDATE_BOARD = "UPDATE_BOARD";
export const UPDATE_GAME = "UPDATE_GAME";
export const UPDATE_GAME_MODE = "UPDATE_GAME_MODE";
export const UPDATE_SPECTRES = "UPDATE_SPECTRES";

export const DISPLAY_TOAST = "DISPLAY_TOAST";

export const CONNECTION = "connection";
export const DISCONNECT = "disconnect";

export const ADD_CHAT_MESSAGE = "ADD_CHAT_MESSAGE";
export const NEW_CHAT_MESSAGE = "NEW_CHAT_MESSAGE";

export const KEY_PRESSED = "KEY_PRESSED";
