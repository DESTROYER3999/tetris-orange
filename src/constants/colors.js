export const MALUS = "#3d3d3d";

export const VIVID_BLUE = "#143cdc";

export const CYAN_BLUE = "#14d4dc";

export const CRIMSON = "#dc143c";

export const PURE_ORANGE = "#ff8b00";

export const LIME_GREEN = "#3cdc14";

export const PURE_VIOLET = "#b600ff";

export const PURE_YELLOW = "#fff400";

export const DARK_CYAN = "#375f64";

export const GREEN_CHAT = "#73F054";

export const RED_CHAT = "#ff3366";
