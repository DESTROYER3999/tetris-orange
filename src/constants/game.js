export const SOLO = "Solo";
export const BATTLEROYAL = "Battleroyal";
export const MAX_PLAYER_SOLO = 1;
export const MAX_PLAYER_BATTLEROYAL = 5;
export const BOARD_HEIGHT = 20;
export const BOARD_WIDTH = 10;
