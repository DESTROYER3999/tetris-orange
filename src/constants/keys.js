export const enumKeys = {
  ARROW_LEFT: "ArrowLeft",
  ARROW_RIGHT: "ArrowRight",
  ARROW_DOWN: "ArrowDown",
  ARROW_UP: "ArrowUp",
  SPACE: "Space",
  ENTER: "Enter"
};
