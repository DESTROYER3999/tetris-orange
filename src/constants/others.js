export const LOG_LINE =
  "\n_________________________________________________________________________________________________________\n\n";
