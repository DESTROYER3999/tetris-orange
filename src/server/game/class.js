import { updateHighScoresInDb } from "../../database";
class Game {
  constructor() {
    this._rooms = [];
    this._players = [];
    this._highscores = [
      {
        score: 0,
        name: ""
      },
      {
        score: 0,
        name: ""
      },
      {
        score: 0,
        name: ""
      },
      {
        score: 0,
        name: ""
      },
      {
        score: 0,
        name: ""
      }
    ];
  }

  get rooms() {
    return this._rooms;
  }

  get players() {
    return this._players;
  }
  get highscores() {
    return this._highscores;
  }

  set highscores(highscores) {
    this._highscores = highscores;
  }

  addRoom(room) {
    this._rooms.push(room);
  }

  removeRoom(roomName) {
    this._rooms = this._rooms.filter(room => room.name !== roomName);
  }

  addPlayer(player) {
    this._players.push(player);
    return player;
  }

  removePlayer(playerId) {
    this._players = this._players.filter(player => player.id !== playerId);
  }

  findRoom(roomName) {
    return this._rooms.find(room => room.name === roomName);
  }

  findPlayer(playerId) {
    return this._players.find(player => player.id === playerId);
  }

  updateHighScores(score, playerName) {
    const newHighScore = { score, playerName };
    this._highscores.every((highscore, index) => {
      if (highscore.score < score) {
        this._highscores.splice(index, 0, newHighScore);
        this._highscores.length = 5;
        updateHighScoresInDb(this._highscores);
        return false;
      } else return true;
    });
  }

  createPublicRoomsArray() {
    return this._rooms.map(room => ({
      name: room.name,
      playersCount: room.playersCount,
      mode: room.mode,
      isStarted: room.isStarted
    }));
  }

  toObject() {
    return {
      rooms: this.createPublicRoomsArray(),
      highscores: this._highscores
    };
  }
}

const instance = new Game();

export default instance;
