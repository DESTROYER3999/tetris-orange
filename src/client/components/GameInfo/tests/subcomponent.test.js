import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GameInfoSub from "../subcomponent";
import socket from "../../../services/socket-api";
import { SOLO, BATTLEROYAL } from "../../../../constants/game";
import Select from "../../common/Select";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

const props = {
  mode: SOLO,
  playersCount: 1,
  name: "Player1",
  isStarted: false,
  isHost: true,
  onChangeGameMode: jest.fn()
};

describe("<GameInfoSub />", () => {
  test("GameInfoSub rendering in solo mode", () => {
    const wrapper = shallow(<GameInfoSub {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("GameInfoSub rendering in battleroyal mode", () => {
    props.mode = BATTLEROYAL;
    const wrapper = shallow(<GameInfoSub {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("GameInfoSub call onChangeGameMode when the mode selector change", () => {
    props.mode = SOLO;
    const wrapper = shallow(<GameInfoSub {...props} />);
    const newMode = BATTLEROYAL;
    const event = { target: { value: newMode } };

    wrapper.find(Select).simulate("change", event);
    expect(props.onChangeGameMode).toHaveBeenCalled();
  });
});
