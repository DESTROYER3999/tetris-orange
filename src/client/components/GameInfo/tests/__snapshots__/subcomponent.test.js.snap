// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfoSub /> GameInfoSub rendering in battleroyal mode 1`] = `
<styled.div
  id="game-info"
>
  <styled.div>
    Player1
  </styled.div>
  <styled.div>
    <SelectMenu
      disabled={false}
      label="Mode"
      onChange={[Function]}
      value="Battleroyal"
    >
      <WithStyles(WithStyles(ForwardRef(MenuItem)))
        disabled={false}
        value="Solo"
      >
        Solo
      </WithStyles(WithStyles(ForwardRef(MenuItem)))>
      <WithStyles(WithStyles(ForwardRef(MenuItem)))
        value="Battleroyal"
      >
        Battleroyal
      </WithStyles(WithStyles(ForwardRef(MenuItem)))>
    </SelectMenu>
    <div
      className="description"
    >
      <styled.label>
        Description
      </styled.label>
      <styled.p>
        Multiplayer mode, be the last one alive to win ! Removing multiple lines at a time send malus blocks to your opponents.
      </styled.p>
    </div>
  </styled.div>
  <div
    className="player-count"
  >
    [
    1
    /
    5
     Players]
  </div>
</styled.div>
`;

exports[`<GameInfoSub /> GameInfoSub rendering in solo mode 1`] = `
<styled.div
  id="game-info"
>
  <styled.div>
    Player1
  </styled.div>
  <styled.div>
    <SelectMenu
      disabled={false}
      label="Mode"
      onChange={[Function]}
      value="Solo"
    >
      <WithStyles(WithStyles(ForwardRef(MenuItem)))
        disabled={false}
        value="Solo"
      >
        Solo
      </WithStyles(WithStyles(ForwardRef(MenuItem)))>
      <WithStyles(WithStyles(ForwardRef(MenuItem)))
        value="Battleroyal"
      >
        Battleroyal
      </WithStyles(WithStyles(ForwardRef(MenuItem)))>
    </SelectMenu>
    <div
      className="description"
    >
      <styled.label>
        Description
      </styled.label>
      <styled.p>
        Single player mode, try to reach the highest score !
      </styled.p>
    </div>
  </styled.div>
  <div
    className="player-count"
  >
    [
    1
    /
    1
     Players]
  </div>
</styled.div>
`;
